### ISSUE TITLE
Associating a probability to a hit belonging to one or more tracks

### ISSUE THREAD

**Issue by SwapneelM** [SwapneelM/DeepHGCal#4]
As per the Tensorflow documentation:

**If using exclusive labels (wherein one and only one class is true at a time), see sparse_softmax_cross_entropy_with_logits**.

We presently stick with the [tf.nn.sparse_softmax_cross_entropy_with_logits](https://www.tensorflow.org/api_docs/python/tf/nn/sparse_softmax_cross_entropy_with_logits) loss function since we associate a rechit to only one single track even though it might belong to multiple tracks. There is, however, scope to do this using a different loss function already available in Tensorflow.

This allows us to consider a case where a rechit can belong to multiple tracks with varying probabilities if we use the [tf.nn.softmax_cross_entropy_with_logits_v2](https://www.tensorflow.org/api_docs/python/tf/nn/softmax_cross_entropy_with_logits_v2) function for calculating the loss.

This could prove an interesting part of future work on tracking in DeepHGCal.
